src/hal/src/pso/descriptor.rs

```diff
@@ -39,7 +39,7 @@ pub type DescriptorArrayIndex = usize;
 #[repr(C)]
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub enum DescriptorType {
+pub enum RawDescriptorType {
```

Comment:
Let's remove it entirely :)

In Dx11 we can just rely on `Ord` implementation for our `DescriptorType`.
In Vulkan we should be having the `match` into `ash` descriptor type (which matches this one).